src/transformers/modeling_squeezebert.py

```diff
@@ -0,0 +1,1032 @@
+# coding=utf-8
+# Copyright 2020 The SqueezeBert authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch SqueezeBert model. """
+
+
+import logging
+import math
+
+import torch
+from torch import nn
+from torch.nn import CrossEntropyLoss, MSELoss
+
+from .activations import gelu
+from .configuration_squeezebert import SqueezeBertConfig
+from .file_utils import add_code_sample_docstrings, add_start_docstrings, add_start_docstrings_to_callable
+from .modeling_outputs import (
+    BaseModelOutput,
+    BaseModelOutputWithPooling,
+    MaskedLMOutput,
+    MultipleChoiceModelOutput,
+    QuestionAnsweringModelOutput,
+    SequenceClassifierOutput,
+    TokenClassifierOutput,
+)
+from .modeling_utils import PreTrainedModel
+
+
+logger = logging.getLogger(__name__)
+
+_CONFIG_FOR_DOC = "SqueezeBertConfig"
+_TOKENIZER_FOR_DOC = "SqueezeBertTokenizer"
+
+####################################################
+# This list contrains shortcut names for some of
+# the pretrained weights provided with the models
+####################################################
+SQUEEZEBERT_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "squeezebert/squeezebert-uncased",
+    "squeezebert/squeezebert-mnli",
+    "squeezebert/squeezebert-mnli-headless",
+]
+
+
+def transpose_x(x):
```

Comment:
nit. `def transpose` is a better name than `def transpose_x` I think

Reply:
Actually I don't really think this abstraction is needed at all. It's a one-liner `x.permute(0, 2, 1)` and doesn't really need a function abstraction IMO.

Reply:
Also, `x.permute(0, 2, 1)` is used directly quite often below ;-)